pkg/infrastructure/powervs/clusterapi/powervs.go

```diff
@@ -98,118 +95,114 @@ func (p Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput)
 	}
 	logrus.Debugf("InfraReady: image = %+v", powerVSImage)
 
-	// SAD: client in the Metadata struct is lowercase and therefore private
-	// client = in.InstallConfig.PowerVS.client
-	client, err = powervsconfig.NewClient()
-	if err != nil {
-		return fmt.Errorf("failed to get NewClient in InfraReady: %w", err)
-	}
-	logrus.Debugf("InfraReady: NewClient returns %+v", client)
-
 	// We need to set the region we will eventually query inside
-	vpcRegion = in.InstallConfig.Config.Platform.PowerVS.VPCRegion
+	vpcRegion := in.InstallConfig.Config.Platform.PowerVS.VPCRegion
 	if vpcRegion == "" {
 		vpcRegion, err = powervstypes.VPCRegionForPowerVSRegion(in.InstallConfig.Config.Platform.PowerVS.Region)
 		if err != nil {
 			return fmt.Errorf("failed to get VPC region (%s) in InfraReady: %w", vpcRegion, err)
 		}
 	}
 	logrus.Debugf("InfraReady: vpcRegion = %s", vpcRegion)
-	if err = client.SetVPCServiceURLForRegion(ctx, vpcRegion); err != nil {
+	if err = in.InstallConfig.PowerVS.SetVPCServiceURLForRegion(ctx, vpcRegion); err != nil {
 		return fmt.Errorf("failed to set the VPC service region (%s) in InfraReady: %w", vpcRegion, err)
 	}
 
-	// Step 1.
-	// Create DNS records for the two load balancers
-	// map[string]VPCLoadBalancerStatus
-	instanceCRN, err = client.GetInstanceCRNByName(ctx,
-		in.InstallConfig.PowerVS.BaseDomain,
-		in.InstallConfig.Config.Publish)
+	// Step 1: Create DNS records for the two load balancers
+	if err = createLoadBalancerDNSRecords(ctx, in, powerVSCluster.Status.LoadBalancers); err != nil {
+		return fmt.Errorf("failed to create DNS records for loadbalancers: %w", err)
+	}
+
+	// Step 2: See which ports are already allowed.
+	missingPorts, err := findMissingSecurityGroupRules(ctx, in, *powerVSCluster.Status.VPC.ID)
+	if err != nil {
+		return fmt.Errorf("failed to find missing security group rules: %w", err)
+	}
+
+	// Step 3: Add to security group rules
+	for port := range missingPorts {
```

Comment:
```suggestion
	for port := range missingPorts {
	    port := port // TODO: remove when using golang 1.22+
```
Loop variable is reused up to golang 1.21 so without a local copy you're always pointing to the same location.